src/systems/si-hep/include/units/isq/si/hep/area.h

```diff
@@ -0,0 +1,104 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2021 Mateusz Pusz
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+// IWYU pragma: begin_exports
+#include <units/isq/dimensions/area.h>
+#include <units/quantity.h>
+#include <units/reference.h>
+#include <units/symbol_text.h>
+// IWYU pragma: end_exports
+
+#include <units/isq/si/length.h>
+#include <units/unit.h>
+
+namespace units::isq::si::hep {
+
+// effective cross-sectional area according to EU council directive 80/181/EEC
+// https://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=CELEX:01980L0181-20090527#page=10
+// https://www.fedlex.admin.ch/eli/cc/1994/3109_3109_3109/de
+struct barn : named_scaled_unit<barn, "b", prefix, ratio(1, 1, -28), square_metre> {};
+struct yocto_barn : prefixed_unit<yocto_barn, yocto, barn> {};
+struct zepto_barn : prefixed_unit<zepto_barn, zepto, barn> {};
+struct atto_barn : prefixed_unit<atto_barn, atto, barn> {};
+struct femto_barn : prefixed_unit<femto_barn, femto, barn> {};
+struct pico_barn : prefixed_unit<pico_barn, pico, barn> {};
+struct nano_barn : prefixed_unit<nano_barn, nano, barn> {};
+struct micro_barn : prefixed_unit<micro_barn, micro, barn> {};
+struct milli_barn : prefixed_unit<milli_barn, milli, barn> {};
+
+//template<UnitOf<dim_area> U, Representation Rep = double>
+//  using area = quantity<dim_area, U, Rep>;
```

Comment:
simply commented/dead-code... 

This was initially needed but I preferred the existing `square_metre` definition and added
```cpp 
#include <units/isq/dimensions/area.h>
```
I subsequently commented it to avoid the redundant definition/collision. It's now removed. 

N.B. the SI HEP extension are basically only additional scaling definitions and otherwise behave like the rest of SI.

Reply:
So why other files still have their own definitions? For example:

```cpp
template<UnitOf<dim_energy> U, Representation Rep = double>
using energy = quantity<dim_energy, U, Rep>;
```

Reply:
Maybe you wanted `dim_energy` to be in `eV`?

Reply:
I think you probably missed my previous comment above?

Reply:
> Maybe you wanted `dim_energy` to be in `eV`?

Sorry for not commenting on this properly. mea culpa

Basically yes. While the 'eV' is a straight scaling of 'J', the aim was to avoid numeric errors such as underflow, loss of numeric precision, etc. for most of the computation which is typically done in 'eV' due to the intrinsic large difference in orders-of-magnitudes (ie. 1e-19). The conversion to Joules would then be done only in places where it is explicitly necessary (e.g. when sending data to physical actuators/controllers).

Hope this rationale is sound enough. 

I found some related issues for the unit tests regarding the barn (1e28 orders-of-magnitude different to sqm) when checking that 1e-28 sqm equals 1 barn and vice versa. The results confirmed the identity from a practical point-of-view, but numerically there were rounding errors on the level of the last bit of the 'double' mantissa. I commented/removed those unit-test lines because I felt that these were possibly beyond the scope (ie. > 64-bit floating-point math) of this library. Hope I didn't misjudge this.

Let me know if I should make further changes on the present state.

Reply:
I meant I do not see any `dim_energy` in the `hep` namespace. Probably you wanted to add some...

Reply:
I thought of reusing the one in
```cpp
#include <units/isq/si/energy.h>
```

because they are closely linked, to avoid redefinitions or collisions (e.g. force, length), and since you asked not to redefine the `electronvolt' but to reuse the one in the 'si' header).  I tried the following simple definition of
```cpp
struct dim_energy : isq::dim_energy<dim_energy, electronvolt, dim_force, dim_length> {};
```
which yields issues w.r.t. the `downcast_guide`. Reusing the version from 'si' seems to work fine ... or am I missing something?

Do you have a better option in mind?

Reply:
I know that there might be some downcasting-related issues. On the other hand, you redefine dimensions for `dim_mass` and `dim_momentum` with units different than SI. I found it inconsistent and wanted to ensure that everything is OK.

Reply:
True, the difference for `dim_mass` and `dim_momentum` appears to be that their corresponding header files have their own `named_scalled_unit` definitions that have no equivalence in the existing 'SI' definitions. The SI energy header had existing `eV`, `GeV`, `electronvolt`, and 'gigaelectronvolt` definitions that needed to be kept in 'SI'.

I have a momentary lack of clear vision on this. What would be your preference/proposal to move forward?

Reply:
I am fine with merging it as it is now and possibly change it later on on the go if needed. I just wanted to ensure that everything is OK and consistent.